src/shaderlib/fp64/fp64.js

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2015 - 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import fp64arithmeticShader from './fp64-arithmetic.glsl';
+import fp64functionShader from './fp64-functions.glsl';
+
+const fp64shader = `${fp64arithmeticShader}\n${fp64functionShader}`;
+
+export default {
+  name: 'fp64',
+  vs: fp64shader,
+  fs: null
+};
+
+// Arithmetic only
+export const fp64arithmetic = {
+  name: 'fp64-arithmetic',
+  vs: `${fp64arithmeticShader}`,
+  fs: null
+};
+
+// Fragment shader fp64
+export const fp64fs = {
+  name: 'fp64-fs',
+  vs: null,
+  fs: fp64shader
+};
```

Comment:
It is actually very similar, the big difference is support for `dependencies` and more advanced `getUniforms`. I just opened https://github.com/uber/luma.gl/pull/258, with hopefully better documentation of the new shader module system.

About `fp64fs` - The complication here is that many shader modules are matched pairs of fragment and vertex shaders, but some like `fp64` can be useful as a general library either in vertex or fragment (e.g. your 64 bit Mandelbrot example could use `fp64fs`).

I admit it is a little clunky, I think we may be able to come up with a better API for this.